cpp/log/sqlite_db-inl.h

```diff
@@ -317,9 +328,27 @@ template <class Logged>
 int64_t SQLiteDB<Logged>::TreeSize() const {
   cert_trans::ScopedLatency latency(
       latency_by_op_ms.GetScopedLatency("tree_size"));
-  std::lock_guard<std::mutex> lock(lock_);
+  std::unique_lock<std::mutex> lock(lock_);
 
   CHECK_GE(tree_size_, 0);
+  sqlite::Statement statement(
```

Comment:
Yeah, but "not cheap" because blocking and "not cheap" because blocking and extending the lock while burning cycles doing SQL on every query are quite different.  Maybe it's ok if you can cache the staleness state as you suggest, but it'd be nice to have some benchmarks showing what effect this has (if any) on our current state of affairs, doesn't need to be over the full 6M certs.